### ISSUE TITLE
Unnecessary import suggestion

### ISSUE THREAD

**Issue by pwlodarczyk92** [intellij-rust/intellij-rust#4983]
## Environment

* **IntelliJ Rust plugin version: 0.2.115.2157-193**
* **Rust toolchain version: 1.40.0**
* **IDE name and version: IntelliJ IDEA 2019.3.2 (Ultimate Edition)**
* **Operating system: Ubuntu 18.04 LTS**

## Problem description
Unnecessary import is suggested for filter method when both diesel and itertools are listed as dependencies.

## Steps to reproduce
dependencies:
```
diesel = "=1.4.3"
itertools = "=0.8.2"
```
`main.rs`:
```
use itertools::Itertools;

fn main() {
    let undivisible_sequences: Vec<Vec<i32>> = (0..10)
        .group_by(|digit| digit % 4 == 0)
        .into_iter()
        .filter(|(divisible, _)| !*divisible)
        .map(|(_, undivisible)| undivisible.collect_vec())
        .collect_vec();
}
```
![unimport](https://user-images.githubusercontent.com/8101990/74636185-c792f600-5167-11ea-9d82-1bf593fecd8d.png)
Import is not suggested and no errors are reported when diesel is not listed as a dependency.

**Comment by Undin** [comment]
Thanks for the report!
Really, plugin fails to infer type of `(0..10).group_by(|digit| digit % 4 == 0).into_iter()` expression. It leads to unresolved `filter` method.

The plugin has `Unresolved reference` inspection to highlight such things. But for now, it highlights unresolved references only when there are suitable things to import to avoid a lot of false-positive annotations (such behavior can be configured by `Ignore unresolved references without quick fix` option in `Preferences | Editor | Inspections | Rust | Unresolved reference`).

So without additional dependencies, the plugin just doesn't show this unresolved `filter` method. But when you add `diesel` dependency, it brings `filter` method with some traits that's why the plugin starts to highlight it

**Comment by Undin** [comment]
@vlad20012 could you take a look?
